openmdao/solvers/linear/tests/test_petsc_ksp.py

```diff
@@ -333,6 +333,84 @@ def test_solve_on_subsystem(self):
         assert_rel_error(self, output[1], g1.expected_solution[0], 3e-15)
         assert_rel_error(self, output[5], g1.expected_solution[1], 3e-15)
 
+    def test_linear_solution_cache(self):
+        # Test derivatives across a converged Sellar model.
+
+        class Comp(ImplicitComponent):
```

Comment:
why is this class definition copied in two tests? Pull it out and define it in one place?

Reply:
How strongly do you feel about the need for a defective component that is used twice to show up in our library of testing components? That library is already packed with stuff, and not very well sorted or documented, to the extent that I had to create a new component for this test because I couldn't find what I needed, though an adequate component may have been there. If it is important, than we need a story to go through our test and reorganize things, getting rid of all duplicate testing components, or generalizing and consolidating them into a well-documented taxonomy of components/systems/problems.

Reply:
but anyway, done, and new code pushed up for re-review.